src/reducers/index.js

```diff
@@ -16,22 +15,25 @@ const propertySelector = (state, rtId, id) => state[rtId][id]
 
 export const getProperty = createSelector(
   [resourceTemplateSelector, propertySelector],
-  (resourceTemplate, propertyURI)  => {
+  (resourceTemplate, propertyURI) => {
     return resourceTemplate[propertyURI]
   }
 )
 
+export const setResourceTemplate = async (state, action) => {
```

Comment:
The action payload is a promise now, so use async/await to resolve it to a value.